gateway/src/main/kotlin/Event.kt

```diff
@@ -633,3 +667,37 @@ data class ApplicationCommandUpdate(val application: DiscordApplicationCommand,
 @KordPreview
 data class ApplicationCommandDelete(val application: DiscordApplicationCommand, override val sequence: Int?) :
     DispatchEvent()
+
+data class ThreadCreate(val channel: DiscordChannel, override val sequence: Int?) : DispatchEvent()
+
+data class ThreadUpdate(val channel: DiscordChannel, override val sequence: Int?) : DispatchEvent()
+
+data class ThreadDelete(val channel: DiscordChannel, override val sequence: Int?) : DispatchEvent()
+
+data class ThreadMemberUpdate(val member: DiscordThreadMember, override val sequence: Int?) : DispatchEvent()
+
+data class ThreadListSync(val sync: DiscordThreadListSync, override val sequence: Int?) : DispatchEvent()
+
+data class ThreadMembersUpdate(val members: DiscordThreadMembersUpdate, override val sequence: Int?) : DispatchEvent()
+
+@Serializable
+data class DiscordThreadListSync(
+    @SerialName("guild_id")
+    val guildId: Snowflake,
+    @SerialName("channel_ids")
+    val channelIds: List<Snowflake>,
```

Comment:
This should be optional: https://discord.com/developers/docs/topics/gateway#thread-list-sync